packages/pages/src/vite-plugin/build/build.ts

```diff
@@ -45,31 +46,46 @@ export const build = (projectStructure: ProjectStructure): Plugin => {
 };
 
 /**
- * Produces a {@link InputOption} by first adding all templates at {@link templateDir} to be output
- * at {@code server/}. Also adds an additional entry-point for all templates ending in tsx to be
- * used to hydrate the bundle.
+ * Produces a {@link InputOption} by adding all templates at {@link templateDir} and
+ * {@link templatedomainDir} to be output at {@code server/}. If there are two files
+ * that share the same name between the two provided template folders, only the file
+ * in domain template folder path is included. Also adds an additional entry-point
+ * for all templates ending in tsx to be used to hydrate the bundle.
  *
- * @param templateDir the directory where templates are stored.
+ * @param templateDir the directory where all templates are stored.
+ * @param templatedomainDir the directory where templates of a specific domain are stored.
  * @param hydrationOutputDir the directory where hydration inputs will be generated at.
  * @returns
  */
 const discoverInputs = async (
-  templateDir: string,
+  templateRootDir: string,
+  templatedomainDir: string | undefined,
   hydrationOutputDir: string
 ): Promise<InputOption> => {
-  return (await readdir(templateDir)).reduce(
-    (input: Record<any, any>, template) => {
-      const parsedPath = parse(template);
+  const entryPoints: Record<string, string> = {};
+  const updateInput = async (dir: string) =>
+    (await readdir(dir, { withFileTypes: true }))
+      .filter((dirent) => !dirent.isDirectory())
+      .map((file) => file.name)
+      .reduce((input, template) => {
```

Comment:
it feels weird to use a reduce here when we're not using the return value directly and are actually modifying the entryPoints object. Maybe a forEach if we want to do it this way?

Reply:
sure